core/lib/object_store/src/gcs_public.rs

```diff
@@ -0,0 +1,163 @@
+use std::time::Duration;
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use http::StatusCode;
+use reqwest::Client;
+
+use crate::{raw::BoxedError, Bucket, ObjectStore, ObjectStoreError};
+
+#[derive(Debug)]
+pub struct PublicReadOnlyGoogleCloudStorage {
+    gcs_base_url: String,
+    bucket_prefix: String,
+    max_retries: u16,
+    client: Client,
+    backoff_seconds: u64,
+}
+
+impl PublicReadOnlyGoogleCloudStorage {
+    pub fn new(
+        gcs_base_url: String,
+        bucket_prefix: String,
+        max_retries: u16,
+        timeout: u64,
+        backoff_seconds: u64,
```

Comment:
Nit: It generally makes sense to supply `Duration`s in such cases, so that they are more configurable, have more conventional `Debug` representation etc.